manifests/write_json.pp

```diff
@@ -0,0 +1,73 @@
+# = Define: sensu::write_json
+#
+# Writes arbitrary hash data to a config file. Note: you must manually notify
+# any Sensu services to restart them when using this defined resource type.
+#
+# Example:
+#
+#  sensu::write_json{'/etc/sensu/conf.d/check.json':
+#    content => {"config" => {"key" => "value"}},
+#    notify  => [Service['sensu-client'], Service['sensu-server']],
+#  }
+#
+# == Parameters
+#
+# [*name*]
+#   String. The config file target path.
+#
+# [*ensure*]
+#   String. Whether the file should be present or not.
+#   Default: present
+#   Valid values: present, absent
+#
+# [*mode*]
+#   String. The file mode.
+#   Default: 0755
+#
+# [*owner*]
+#   String. The file owner.
+#   Default: $sensu::user
+#
+# [*group*]
+#   String. The file group.
+#   Default: $sensu::group
+#
+# [*pretty*]
+#   Boolean. Write the json with "pretty" indenting & formating.
+#   Default: true
+#
+# [*content*]
+#   Hash. The hash content that will be converted to json and written into
+#         the target config file.
+#   Default: {}
+#
+define sensu::write_json(
+  Enum['present', 'absent'] $ensure = 'present',
+  Optional[String]          $mode = '0755',
```

Comment:
I don't think this should be optional after you refactor. Could you order it after group, so it reads owner/group/mode. Same in the file resource below.